File: Programmers/dnjfqhd12345/week_1/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    int a;
    int b;
    int size;
    int idx;
    vector<int> copyarr;
    vector<int> answer;
    for (int i = 0; i < commands.size(); i++) {
        a = commands[i][0];
        b = commands[i][1];
        idx = commands[i][2];
        size = b - a + 1;
        for (int j = a; j <= b; j++) {
            copyarr.push_back(array[j-1]);
        }
        sort(copyarr.begin(), copyarr.end());
        answer.push_back(copyarr[idx-1]);
        copyarr.clear();
    }
    return answer;
}

File: Programmers/k번째 수.py
def solution(array, commands):
    arr = array # [1, 5, 2, 6, 3, 7, 4]
    answer = []
    for i in range(len(commands)):
        step_one_arr = arr[commands[i][0]-1:commands[i][1]] # [5, 2, 6, 3]
        print(step_one_arr)
        step_one_arr.sort() # 2, 3, 5, 6
        print(step_one_arr)
        answer.append(step_one_arr[commands[i][2]-1]) # 5 추가
    
    return answer

